pyensembl/database.py

```diff
@@ -81,25 +82,42 @@ def _create_database(self):
             indices=indices)
         return db
 
-    def _connect_or_create_database(self):
+    def connection_if_exists(self):
         """
-        If database already exists, open a connection.
-        Otherwise, create it.
+        Return the connection if the DB exists, and otherwise return 
+        None.
         """
         db_path = self.local_db_path()
         if exists(db_path):
-            db = sqlite3.connect(db_path)
+            connection = sqlite3.connect(db_path)
             # maybe file got created but not filled
-            if datacache.db.db_table_exists(db, 'ensembl'):
-                return db
-        return self._create_database()
+            if datacache.db.db_table_exists(connection, 'ensembl'):
+                return connection
+        return None
+
+    def _connect_or_create_database(self):
+        """
+        If database already exists, open a connection.
+        Otherwise, create it if auto download is enabled.
+        """
+        connection = self.connection_if_exists()
+        if connection:
+            return connection
+        if self.auto_download:
```

Comment:
I was actually just worrying about that. Changing the database schema is
high on my TODO list, it would be annoying to have to download to
regenerate the database. Any ideas? Maybe purely local operations can still
happen automatically/magically, with a loud warning to the user?
On Feb 12, 2015 11:01 PM, "Tavi Nathanson" <notifications@github.com> wrote:

> In pyensembl/database.py
> <https://github.com/hammerlab/pyensembl/pull/34#discussion_r24643204>:
>
> > -            if datacache.db.db_table_exists(db, 'ensembl'):
> > -                return db
> > -        return self._create_database()
> > +            if datacache.db.db_table_exists(connection, 'ensembl'):
> > +                return connection
> > +        return None
> > +
> > +    def _connect_or_create_database(self):
> > +        """
> > +        If database already exists, open a connection.
> > +        Otherwise, create it if auto download is enabled.
> > +        """
> > +        connection = self.connection_if_exists()
> > +        if connection:
> > +            return connection
> > +        if self.auto_download:
>
> Good idea. What about situations where we want to force a SQLite setup
> without re-downloading? That's probably overcomplicating things.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hammerlab/pyensembl/pull/34/files#r24643204>.
>